src/frequenz/sdk/timeseries/_formula_engine/_formula_engine_pool.py

```diff
@@ -47,14 +47,14 @@ def __init__(
         self._namespace = namespace
         self._channel_registry = channel_registry
         self._resampler_subscription_sender = resampler_subscription_sender
-        self._engines: dict[str, "FormulaEngine|FormulaEngine3Phase"] = {}
+        self._engines: dict[str, "FormulaEngine[Any]|FormulaEngine3Phase[Any]"] = {}
 
     def from_string(
         self,
         formula: str,
         component_metric_id: ComponentMetricId,
         nones_are_zeros: bool = False,
-    ) -> FormulaEngine:
+    ) -> FormulaEngine[Quantity]:
```

Comment:
Also I don't understand why sometimes is `Quantity` and sometimes `Any`.

Reply:
I think `Any` is used only in an intermediate commit,  only in the FormulaEnginePool's storage,  which was very weakly typed,  and using `Quantity` was causing errors that I didn't want to deal with in that commit.  I could have gone ahead and made them Quantity anyway, because we're not expecting intermediate commits to pass tests,  I guess I just didn't do it like that.